Wrappers/Python/ccpi/viewer/cli/resample.py

```diff
@@ -0,0 +1,135 @@
+from argparse import ArgumentParser
+
+import yaml
+import schema
+from schema import SchemaError, Schema, Optional
+
+from ccpi.viewer.utils.io import ImageReader, ImageWriter
+
+'''
+This command line tool takes a dataset file and a yaml file as input.
+It resamples or crops the dataset as it reads it in, and then writes
+out the resulting dataset to a file.
+
+Supported file types for reading:
+hdf5, nxs, mha, raw, numpy
+
+Supported file types for writing:
+hdf5, nxs, mha
+
+'''
+
+'''
+EXAMPLE INPUT YAML FILE:
+
+array:
+  shape: (1024,1024,1024)
+  is_fortran: False
+  is_big_endian: True
+  typecode: 'float32'
+  dataset_name: '/entry1/tomo_entry/data/data' # valid for HDF5 and Zarr
+resample:
+  target_size: 100000
+  resample_z: True
+output:
+  file_name: 'this_fname.nxs'
+  format: 'hdf5' # npy, METAImage, NIFTI (or Zarr to come)
```

Comment:
I think the yaml file is good, however I'd like to have a command line tool which can be called from the command line:

```bash 
resample -i input_file.h5 -o resampled.h5 --target_size 100M --resample_z 1
```

This should parse the parameters, read the metadata in the input file, generate the meta.yaml and execute. 

Raw files are the only ones that require lots of parameters and it may be awkward to have a very long command line to specify all of these. In that case we could have a command like `resample --example` which generates an example basic yaml file

Reply:
or we could support 
```bash

resample -i config.yaml
```